From df0e90a621bac9a9290f58113197f1d6e9ba13d7 Mon Sep 17 00:00:00 2001
From: Alex Wang <alex.ts.wang@fii-foxconn.com>
Date: Fri, 22 Aug 2025 14:09:01 +0800
Subject: [PATCH 1/1] SmbpbiSensor: Add range checking for values

Change:
* Add range checking for valid values

Reason:
* Sensor values from the HMC are prone to occasional out-of-range
  readings (either excessively high or low)
* We do not want to consider these values valid, and incorrectly assert
  any thresholds

ref: https://gerrit.openbmc.org/c/openbmc/dbus-sensors/+/83578
Change-Id: I5dac8df43d9b14a5f9b5505bbd90085613821685

---
 src/smbpbi/SmbpbiSensor.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/smbpbi/SmbpbiSensor.cpp b/src/smbpbi/SmbpbiSensor.cpp
index a2583a6..581e3ee 100644
--- a/src/smbpbi/SmbpbiSensor.cpp
+++ b/src/smbpbi/SmbpbiSensor.cpp
@@ -362,7 +362,8 @@ void SmbpbiSensor::waitReadCallback(const boost::system::error_code& ec)
         return;
     }
 
-    if (ret >= 0)
+    if (ret >= 0 && ((temp >= minValue && temp <= maxValue) ||
+                     (sensorType == "Energy" && temp >= minValue)))
     {
         if constexpr (debug)
         {
@@ -372,7 +373,8 @@ void SmbpbiSensor::waitReadCallback(const boost::system::error_code& ec)
     }
     else
     {
-        lg2::error("Invalid read getRegsInfo");
+        lg2::error("Invalid read at offset: {OFFSET} with value: {VALUE}",
+                   "OFFSET", offset, "VALUE", temp);
         incrementError();
     }
     read();
-- 
2.43.0

